### DISCUSSION TITLE
Add Garbage collector callout or include it as a slider to sampler class, so user can call it when needed

### DISCUSSION THREAD

**Discussion by playlogitech** [comfyanonymous/ComfyUI#362] [Ideas]
Right now if you have huge workflow, especially with multiple upscale/downscale, you getting OOM in blink of an eye.
Im not a python programmer, so im not sure if this code is right. But I imported those nodes into my workflow, things goes better now. I call it after every KSampler workout or when I see vram is close to be full.

```
class gcLatentTunnel:
    @classmethod
    def INPUT_TYPES(s):
        return {"required": { "samples": ("LATENT",)}}
    RETURN_TYPES = ("LATENT",)
    FUNCTION = "gcTunnel"

    CATEGORY = "latent"

    def gcTunnel(self, samples):
        s = samples.copy()

        gc.collect()
        torch.cuda.empty_cache()
        torch.cuda.ipc_collect()
        return (s,)
```

![image](https://user-images.githubusercontent.com/121442734/229361025-01b44816-04e0-4ab1-8dc7-10955aebb133.png)

**Comment by comfyanonymous** [comment]
It's already called at the end of every prompt execution: https://github.com/comfyanonymous/ComfyUI/blob/master/execution.py#L204

I don't want to call it that often because it can reduce performance.

**Reply by playlogitech** [reply]
> at the end

mostly like after end of whole workflow
![image](https://user-images.githubusercontent.com/121442734/229367876-10d18404-9727-4ff7-b3e3-0dc725e5a904.png)

you can just add simple node as i mentioned, so user can decide if it needed or not

**Reply by comfyanonymous** [reply]
What's your hardware and can you give me a traceback of when it OOMs?

**Reply by ntdviet** [reply]
@comfyanonymous I always get a OOM right when decoding latents around 1536x1536, since SDXL1 (ComfyUI on Google Colab)
![ảnh](https://github.com/comfyanonymous/ComfyUI/assets/54492570/7b017bba-bcb5-4ed1-a4a5-a9054625dc1f)
I tried the garbage collector idea from @playlogitech to force torch.cuda.empty_cache() and torch.cuda.ipc_collect() everywhere I suspect a memory shortage, as you see in my workflow screenshot 
![ảnh](https://github.com/comfyanonymous/ComfyUI/assets/54492570/b9a51f41-e185-4bf9-9a3d-69865d9e5820)

but its kind of not working. On the surface, I still get a Warning every prompt execution
